File: java-resend-examples/src/main/java/com/resend/examples/Domains.java
package com.resend.examples;

import com.resend.Resend;
import com.resend.services.domains.model.DomainRecord;
import io.github.cdimascio.dotenv.Dotenv;

public class Domains {
    public static void main(String[] args) {
        Dotenv dotenv = Dotenv.configure().ignoreIfMissing().load();

        String apiKey = dotenv.get("RESEND_API_KEY");
        if (apiKey == null || apiKey.isEmpty()) {
            System.err.println("RESEND_API_KEY environment variable is required");
            System.exit(1);
        }

        Resend resend = new Resend(apiKey);

        try {
            // 1. List all domains
            System.out.println("=== Listing Domains ===");
            var domains = resend.domains().list();
            var domainList = domains.getData();
            System.out.println("Found " + domainList.size() + " domain(s)");

            for (var domain : domainList) {
                System.out.println("  - " + domain.getName() + " (status: " + domain.getStatus() + ", id: " + domain.getId() + ")");
            }

            // 2. Get domain details (if any exist)
            if (!domainList.isEmpty()) {
                String domainId = domainList.get(0).getId();

                System.out.println("\n=== Domain Details: " + domainList.get(0).getName() + " ===");
                var domain = resend.domains().get(domainId);

                System.out.println("Name: " + domain.getName());
                System.out.println("Status: " + domain.getStatus());
                System.out.println("Region: " + domain.getRegion());
                System.out.println("Created: " + domain.getCreatedAt());

                var records = domain.getRecords();
                if (records != null && !records.isEmpty()) {
                    System.out.println("\nDNS Records:");
                    for (DomainRecord record : records) {
                        System.out.println("  " + record.getType() + ": " + record.getName() + " -> " + record.getValue());
                    }
                }

                // 3. Verify domain
                System.out.println("\n=== Verifying Domain ===");
                resend.domains().verify(domainId);
                System.out.println("Domain verification initiated!");
            }

            // To create a new domain (uncomment):
            // var createParams = CreateDomainOptions.builder()
            //         .name("mail.example.com")
            //         .region(Region.US_EAST_1)
            //         .build();
            // var newDomain = resend.domains().create(createParams);

            // To delete a domain (uncomment):
            // resend.domains().remove(domainId);

            System.out.println("\nDone!");
        } catch (Exception e) {
            System.err.println("Error: " + e.getMessage());
            System.exit(1);
        }
    }
}

File: dotnet-resend-examples/Examples/Domains.cs
namespace ResendExamples;

using Resend;

public static class Domains
{
    public static async Task RunAsync()
    {
        var apiKey = Environment.GetEnvironmentVariable("RESEND_API_KEY")
            ?? throw new Exception("RESEND_API_KEY environment variable is required");

        var client = ResendClient.Create(apiKey);

        // 1. List all domains
        Console.WriteLine("=== Listing Domains ===");
        var domains = await client.DomainListAsync();
        Console.WriteLine($"Found {domains.Content.Count} domain(s)");

        foreach (var domain in domains.Content)
        {
            Console.WriteLine($"  - {domain.Name} (status: {domain.Status}, id: {domain.Id})");
        }

        // 2. Get domain details (if any exist)
        if (domains.Content.Count > 0)
        {
            var domainId = domains.Content[0].Id;

            Console.WriteLine($"\n=== Domain Details: {domains.Content[0].Name} ===");
            var domain = (await client.DomainRetrieveAsync(domainId)).Content;

            Console.WriteLine($"Name: {domain.Name}");
            Console.WriteLine($"Status: {domain.Status}");
            Console.WriteLine($"Region: {domain.Region}");
            Console.WriteLine($"Created: {domain.MomentCreated}");

            if (domain.Records?.Count > 0)
            {
                Console.WriteLine("\nDNS Records:");
                foreach (var record in domain.Records)
                {
                    Console.WriteLine($"  {record.RecordType}: {record.Name} -> {record.Value}");
                }
            }

            // 3. Verify domain
            Console.WriteLine("\n=== Verifying Domain ===");
            await client.DomainVerifyAsync(domainId);
            Console.WriteLine("Domain verification initiated!");
        }

        // To create a new domain (uncomment):
        // var newDomain = await client.DomainAddAsync(new DomainAddData
        // {
        //     DomainName = "mail.example.com",
        //     Region = DeliveryRegion.UsEast1
        // });

        // To delete a domain (uncomment):
        // await client.DomainDeleteAsync(domainId);

        Console.WriteLine("\nDone!");
    }
}
